ext/json/json_encoder.c

```diff
@@ -104,7 +104,11 @@ static inline void php_json_encode_double(smart_str *buf, double d, int options)
 {
 	size_t len;
 	char num[PHP_JSON_DOUBLE_MAX_LENGTH];
-	php_gcvt(d, (int)EG(precision), '.', 'e', &num[0]);
+	if (JSON_G(precision) == -1) {
+		php_0cvt(d, 17, '.', 'e', num);
+	} else {
+		php_gcvt(d, JSON_G(precision), '.', 'e', num);
+	}
```

Comment:
OK. Sounds very reasonable. I'll change this.

Any comments fixed size buffer max definition? I just followed spprintf()'s code, but there are number of codes that calculate buffer size. I would like to use fixed size buffer for better performance.